disclosures/2023-05-21.md

```diff
@@ -0,0 +1,60 @@
+# Incident disclosure - 2023-05-21
+
+## Summary
+- Yearn's yVault staking zap on Optimism did not properly refund a user attempting to zap in funds above the USDT vault's deposit limit. Excess funds sent to the zap should have been returned to the user, but instead they remained in the zap contract.
+- Once space in the vault opened up for another user to deposit, the next user to zap in received their own expected USDT as well as the "stuck" USDT from the zap contract. This sequence occurred twice.
+- One user benefitted ~$20k, another lost ~$21k, and a third had a net gain of $1k from a loss and gain (and reported the bug). The user whose funds were stuck has been made whole, and the user who reported the issue has been awarded a bug bounty.
+- Vault deposit limits have been raised, and all staking pools have been migrated to a new zap that prevents this issue from happening in the future. Yearn will continue distribution of our OP grant uninterrupted, with over $1 million in rewards remaining. 
```

Comment:
Split into two thoughts:

- In response, Yearn has deployed a new zap that properly handles the edge case. Additionally, deposit limits on vaults have been raised to accommodate additional deposits.
- Yearn will continue distribution of our OP grant uninterrupted, with over $1 million in rewards remaining.